google/cloud/spanner/client_test.cc

```diff
@@ -770,6 +770,57 @@ TEST(ClientTest, CommitMutatorSessionAffinity) {
   EXPECT_EQ(*timestamp, result->commit_timestamp);
 }
 
+TEST(ClientTest, CommitMutatorSessionNotFound) {
+  auto timestamp = internal::TimestampFromString("2019-08-14T21:16:21.123Z");
+  ASSERT_STATUS_OK(timestamp);
+
+  auto conn = std::make_shared<MockConnection>();
+  EXPECT_CALL(*conn, Commit(_))
+      .WillOnce([&timestamp](Connection::CommitParams const& cp) {
+        EXPECT_THAT(cp.transaction, HasSession("session-3"));
+        return CommitResult{*timestamp};
+      });
+
+  int n = 0;
+  auto mutator = [&n](Transaction const& txn) -> StatusOr<Mutations> {
+    SetSessionName(txn, "session-" + std::to_string(++n));
```

Comment:
Should `EXPECT_THAT(txn, DoesNotHaveSession())` to check that we got an unbound transaction before binding the new session name.

Reply:
Done